File: native-micronaut-api/src/main/java/com/xavierbouclet/nativeapi/ActorService.java
package com.xavierbouclet.nativeapi;

import jakarta.inject.Inject;
import jakarta.inject.Singleton;
import java.util.*;

@Singleton
record ActorService(
    @Inject ActorRepository  repository,
    @Inject MovieRepository movieRepository) {

    public Iterable<Actor> list() {
        return repository.findAll();
    }

    public Actor add(Actor actor) {
        return repository.save(actor);
    }

    public Actor modify(Actor actor) {
        return repository.save(actor);
    }

    public UUID delete(UUID id) {
        repository.deleteById(id);
        return id;
    }

    public Actor get(UUID id) {
        return repository.findById(id).get();
    }

    public Actor
    addMovieToActor(UUID id, UUID idMovie) {

        var movie = movieRepository.findById(idMovie).get();
        var actor = repository.findById(id).get();
        movie.getActors().add(actor);
        actor.getMovies().add(movie);
        return repository.save(actor);
    }

    public Actor removeMovieFromActor(UUID id, UUID idMovie) {
        var movie = movieRepository.findById(idMovie).get();
        var actor = repository.findById(id).get();
        movie.getActors().remove(actor);
        actor.getMovies().remove(movie);
        return repository.save(actor);
    }

}

File: native-micronaut-api/src/main/kotlin/com/xavierbouclet/nativeapi/ActorService.kt
package com.xavierbouclet.nativeapi

import jakarta.inject.Inject
import jakarta.inject.Singleton
import java.util.*

@Singleton
class ActorService(
    @Inject val repository: ActorRepository,
    @Inject val movieRepository: MovieRepository) {

    fun list(): MutableIterable<Actor> {

        return repository.findAll()
    }

//    fun list(title: String): List<Actor> {
//        return repository.findByFullName(title)
//    }

    fun add(actor: Actor): Actor {
        return repository.save(actor)
    }

    fun modify(actor: Actor): Actor {
        return repository.save(actor)
    }

    fun delete(id: UUID): UUID {
        repository.deleteById(id)
        return id
    }

    fun get(id: UUID): Actor? {
        return repository.findById(id).orElseGet(null)
    }

    fun addMovieToActor(id: UUID, idMovie: UUID): Actor {
        val movie = movieRepository.findById(idMovie).get()
        val actor = repository.findById(id).get()
        movie.actors?.add(actor)
        actor.movies?.add(movie)
        return repository.save(actor)
    }

    fun removeMovieFromActor(id: UUID, idMovie: UUID): Actor {
        val movie = movieRepository.findById(idMovie).get()
        val actor = repository.findById(id).get()
        movie.actors?.remove(actor)
        actor.movies?.remove(movie)
        return repository.save(actor)
    }

}
